### ISSUE TITLE
zcash-cli subcommands are a mishmash, help is not helping...

### ISSUE THREAD

**Issue by seefood** [zcash/zcash#1436]
The "help" is not helping put an order to the mess of options, and some of the commands don't seem to work. there are z_commands that seem to echo non z_ prefixed ones, but I don't se a document explaining all this either (other than the updates of "Beta guide" Wiki page).

Example:
﻿⁠⁠⁠⁠`zcash-cli backupwallet`﻿ creates a pretty big file⁠⁠⁠⁠ (and it grows every time I mine a block and get a new T-ADDR), I guess it means all my T-addr private keys are in that file?
However, what do ﻿⁠⁠⁠⁠dumpwallet﻿⁠⁠⁠⁠ and ﻿⁠⁠⁠⁠z_exportwallet﻿⁠⁠⁠⁠ supposed to do? They don't create a file. I could not find a clear way to backup and restore the z-addr information.

Another:
Should `sendtoaddress` still be used or is it only a deprecated bitcoin code leftover? is `z_sendmany` breaking with the syntax and functionality of `sendmany` and is actually a more JSON-esque `sendtoaddress`?

And I'm sure there are more old deprecated subcommands and help text to fix...

In short, This is a gap that should be closed so the tool can be adopted. As it stands now, it's so confusing I'm not sure how to fix the phrasings myself, though I'd love to help.

**Comment by daira** [comment]
In general:

The commands without "z_" or "zc"  do not support any functionality of protected notes. They are identical to the Bitcoin equivalents, in order to avoid breaking software dependent on the Bitcoin RPC. We may deprecate some of them, but not before launch.

The commands starting "zc" are deprecated (except for zcbenchmark). They were a low-level interface for testing and development that exposed many protocol details. They will be removed before launch.

The "z_" commands support protected addresses, and also support transparent addresses if that makes sense for the particular command. That is, they're generalisations of the corresponding non-z_ command. If there's a z_ command and a similarly named non-z_ one, then use the z_ one.

**Comment by daira** [comment]
BTW, dumpwallet does not include z-addresses in its output. This is documented. Use z_exportwallet instead.

**Comment by seefood** [comment]
z_exportwallet creates no file, so I guess that's a bug :-)

**Comment by bitcartel** [comment]
Are you sure?  z_exportwallet filename

**Comment by daira** [comment]
By the way, I don't like the fact that ``dumpwallet`` and ``z_exportwallet`` create files at arbitrary paths with the permissions of the zcashd user; it's a privilege escalation flaw waiting to happen. We might change either or both of these calls.
